rmldatatfbbox/rmldatatfbbox/core.py

```diff
@@ -5,77 +5,158 @@
 Core command group and commands for TF Bounding Box dataset plugin.
 """
 import click 
-
-### OPTIONS ###
-
+import sys
+import io
+import importlib
+import shutil
+from pathlib import Path
+import tensorflow as tf
+from rmldatatfbbox.utils.helpers import (construct_all, write_label_map, export_as_TFExample)
+from ravenml.utils.question import user_input, cli_spinner
+from ravenml.data.helpers import default_filter_and_load
+from ravenml.data.write_dataset import write_dataset, write_metadata
+from ravenml.data.options import pass_create
+from ravenml.data.interfaces import CreateInput, CreateOutput
 
 ### COMMANDS ###
-@click.command(help='TensorFlow Object Detection with bounding boxes.')
+@click.group(help='Dataset creation with bounding box info')
 @click.pass_context
 def tf_bbox(ctx):
     pass
+
+@tf_bbox.command(help='Create a dataset.')
+@pass_create
+@click.pass_context
+def create(ctx: click.Context, create: CreateInput):
+
+    cli_spinner("Importing TensorFlow...", _import_od)
```

Comment:
I'm glad to see this function getting used, however in the current setup it is useless since you're importing tensorflow directly on line 13 of `core.py`. Notice that in `import_od` the tensorflow library, along with other OD libraries, are imported. 

This raises the larger point of whether we want to try and keep using this pattern or not. Personally I think we should, because it's incredibly annoying to wait 2-3s for TF to be imported on every command you run with ravenML, even basic data exploration commands.

On the other hand, if even one plugin doesn't use this pattern, it defeats the purpose entirely. If we are truly open-sourcing this it could get hard to enforce the pattern. Personally I think we should try and require this of plugin developers, because I don't think it's very complicated to use `_dynamic_import` and it makes raven way more usable. But i can see both sides

Reply:
I have removed it from the core.py file, but I can't remove it from the helpers.py file without it throwing an error. Is there something I am missing?

Reply:
Shouldn't have to worry about this if https://github.com/autognc/ravenML/pull/43 gets merged, so marking resolved for now.